pkg/kp/kv.go

```diff
@@ -101,6 +104,28 @@ func (c consulStore) GetHealth(service, node string) (WatchResult, error) {
 	return *healthRes, nil
 }
 
+func (c consulStore) GetServiceHealth(service string) (map[string]WatchResult, error) {
+	healthRes := make(map[string]WatchResult)
+	key := healthPath(service, "")
+	res, _, err := c.client.KV().List(key, nil)
+	if err != nil {
+		return healthRes, KVError{Op: "get", Key: key, UnsafeError: err}
+	} else if res == nil {
+		return healthRes, nil
+	}
+	for _, kvp := range res {
+		watch := &WatchResult{}
+		err = json.Unmarshal(kvp.Value, watch)
```

Comment:
you're setting this here but not checking it until after the `for` loop - does this mean that all errors except the very last one will be ignored? Is that OK or should the error check be moved into the `for` loop?